var/spack/repos/builtin/packages/gptune/package.py

```diff
@@ -0,0 +1,175 @@
+            cdir = join_path(self.prefix, 'gptuneclcm')
+            self.run_test('cp', options=['-r', cdir, '.'], work_dir='.')
+            self.run_test('rm', options=['-rf', 'build'], work_dir='.')
+            self.run_test('mv', options=['gptuneclcm', 'build'], work_dir='.')
+
+            with open('{0}/run_env.sh'.format(self.install_test_root), 'w') as envfile:
+                envfile.write('if [[ $NERSC_HOST = "cori" ]]; then\n')
+                envfile.write('    export machine=cori\n')
+                envfile.write('elif [[ $(uname -s) = "Darwin" ]]; then\n')
+                envfile.write('    export machine=mac\n')
+                envfile.write('elif [[ $(dnsdomainname) = ' +
+                              '"summit.olcf.ornl.gov" ]]; then\n')
+                envfile.write('    export machine=summit\n')
+                envfile.write('elif [[ $(cat /etc/os-release | grep "PRETTY_NAME") ==' +
+                              ' *"Ubuntu"* || $(cat /etc/os-release | grep' +
+                              ' "PRETTY_NAME") == *"Debian"* ]]; then\n')
+                envfile.write('    export machine=unknownlinux\n')
+                envfile.write('fi\n')
+                envfile.write('export GPTUNEROOT=$PWD\n')
+                envfile.write('export MPIRUN={0}\n'.format
+                              (which(spec['mpi'].prefix.bin + '/mpirun')))
+                envfile.write('export proc=$(spack arch)\n')
+                envfile.write('export mpi={0}\n'.format(spec['mpi'].name))
+                envfile.write('export compiler={0}\n'.format(comp_name))
+                envfile.write('export nodes={0} \n'.format(self.nodes))
+                envfile.write('export cores={0} \n'.format(self.cores))
+                envfile.write('export ModuleEnv=$machine-$proc-$mpi-$compiler \n')
+                envfile.write('software_json=$(echo ",\\\"software_configuration\\\":' +
+                              '{\\\"' + spec['blas'].name +
+                              '\\\":{\\\"version_split\\\":' +
+                              ' [' + str(spec['blas'].versions).replace('.', ',') +
+                              ']},\\\"' + spec['mpi'].name +
+                              '\\\":{\\\"version_split\\\": [' +
+                              str(spec['mpi'].versions).replace('.', ',') + ']},\\\"' +
+                              spec['scalapack'].name +
+                              '\\\":{\\\"version_split\\\": [' +
+                              str(spec['scalapack'].versions).replace('.', ',') +
+                              ']},\\\"' +
+                              str(comp_name) + '\\\":{\\\"version_split\\\": [' +
+                              str(comp_version) + ']}}") \n')
+                envfile.write('loadable_software_json=$(echo ",\\\"loadable_software_' +
+                              'configurations\\\":{\\\"' + spec['blas'].name +
+                              '\\\":{\\\"version_split\\\": [' +
```

Comment:
If you use raw-strings: `r'...'` you won't have to escape everything

Reply:
Thanks for the tip. It's quite a time-consuming process to change every string properly. I would prefer to keep it as it for now.